internal/ingress/controller/parser/parser.go

```diff
@@ -1682,26 +1715,46 @@ func toKongPlugin(plugin plugin) kong.Plugin {
 	return result
 }
 
-func kongPluginFromK8SClusterPlugin(k8sPlugin configurationv1.KongClusterPlugin) kong.Plugin {
+func (p *Parser) kongPluginFromK8SClusterPlugin(k8sPlugin configurationv1.KongClusterPlugin) (kong.Plugin, error) {
+	var err error
+	if k8sPlugin.ConfigFrom != (configurationv1.NamespacedSecretValueFromSource{}) {
+		if len(k8sPlugin.Config) > 0 {
+			err = errors.Errorf("plugin '%v' has both Config and ConfigFrom set", k8sPlugin.Name)
+		} else {
+			config, configError := p.namespacedSecretToConfiguration(k8sPlugin.ConfigFrom)
+			err = configError
```

Comment:
Wrap this error up in another error using `errors.Wrap` to add context. Something along the lines:
```
errors.Wrap(err, "creating config from secret")
```

Providing context to these errors is very important as they currently only get logged in controller's output.
Eventually, this has to move into `Status` subresource of CRDs, and then this can be simplified.